deploy/docker/api.py

```
@@ -472,20 +529,88 @@ async def handle_crawl_request(
+            # Clean up non-serializable properties
+            cleaned_dict = {}
+            for key, value in result_dict.items():
+                try:
+                    # Test if value is JSON serializable
+                    json.dumps(value)
+                    cleaned_dict[key] = value
+                except (TypeError, ValueError):
+                    # Handle special cases with proper type detection and normalization
+                    if isinstance(value, (datetime, date)):
+                        # Use existing datetime_handler for datetime/date objects
+                        cleaned_dict[key] = datetime_handler(value)
```

Ruff (0.13.1)

Local variable `datetime_handler` referenced before assignment

(F823)